pandas/core/internals.py

```diff
@@ -5143,7 +5223,12 @@ def _vstack(to_stack, dtype):
         return np.vstack(to_stack)
 
 
-def _maybe_compare(a, b, op):
+def _maybe_compare(a, b, regex=False):
+    if not regex:
+        op = lambda x: operator.eq(x, b)
+    else:
```

Comment:
I want to add regex support in _maybe_compare. The comparing behavior is decided by param regex ( whether to use regex match or equality comparison) . The result will be a mask that  will be passed on to _replace_coerce. This can avoid the result of previous round of comparing overwritten in the succeeding ones, e.g in, {"a":"b", "b":"a"} .